src/partners/bity.ts

```diff
@@ -1,94 +1,113 @@
-        return encodeURIComponent(key) + '=' + encodeURIComponent(credentials[key])
-    }).join('&')
+  if (
+    typeof pluginParams.apiKeys.clientId === 'string' &&
+    typeof pluginParams.apiKeys.clientSecret === 'string'
+  ) {
+    credentials = {
+      grant_type: 'client_credentials',
+      scope: 'https://auth.bity.com/scopes/reporting.exchange',
+      client_id: pluginParams.apiKeys.clientId,
+      client_secret: pluginParams.apiKeys.clientSecret
+    }
 
-        const tokenResponse = await fetch(BITY_TOKEN_URL, {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/x-www-form-urlencoded'
-          },
-          body: tokenParams
-        })
-        const tokenReply = await tokenResponse.json()
-        authToken = tokenReply.access_token
+    tokenParams = Object.keys(credentials)
+      .map(key => {
+        return (
+          encodeURIComponent(key) + '=' + encodeURIComponent(credentials[key])
+        )
+      })
+      .join('&')
 
+    const tokenResponse = await fetch(BITY_TOKEN_URL, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded'
+      },
+      body: tokenParams
+    })
+    const tokenReply = await tokenResponse.json()
+    authToken = tokenReply.access_token
   } else {
     return {
       settings: { lastCheckedMonth: queryMonth, lastCheckedYear: queryYear },
       transactions: []
     }
   }
 
-  let beforeCurrentMonth = true;
-  let currentDate = new Date(Date.now())
+  let beforeCurrentMonth = true
+  const currentDate = new Date(Date.now())
   let currentMonth = (currentDate.getMonth() + 1).toString()
-  currentMonth = currentMonth.length === 1 ? 0 + currentMonth : currentMonth
-  let currentYear = (currentDate.getFullYear()).toString()
+  currentMonth = currentMonth.length === 1 ? `0${currentMonth}` : currentMonth
+  let currentYear = currentDate.getFullYear().toString()
   while (beforeCurrentMonth) {
     if (queryMonth === currentMonth && queryYear === currentYear) {
-      beforeCurrentMonth = false;
+      beforeCurrentMonth = false
     }
     let moreCurrentMonthsTransactions = true
     let page = 1
-    while(moreCurrentMonthsTransactions) {
-      let monthlyTxs: ReturnType<typeof asBityResult>
-      monthlyTxs = []
+    while (moreCurrentMonthsTransactions) {
+      let monthlyTxs
       try {
-        const monthlyResponse = await fetch(`${BITY_API_URL}${queryYear}-${queryMonth}/orders?page=${page}`,
+        const monthlyResponse = await fetch(
+          `${BITY_API_URL}${queryYear}-${queryMonth}/orders?page=${page}`,
           {
             method: 'GET',
             headers: { Authorization: `Bearer ${authToken}` }
-          })
-      
-        if (monthlyResponse.ok) {
-          monthlyTxs = asBityResult(await monthlyResponse.json().catch(e => []))
-        }
+          }
+        )
+        monthlyTxs = asBityResult(await monthlyResponse.json().catch(e => []))
```

Comment:
put back in monthlyresponse.ok, write an else to return an empty array if an error

Reply:
look at specific error code for invalid page number